common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -314,6 +322,57 @@ public static String simpleClassName(Class<?> clazz) {
         }
     }
 
+    /**
+     * Escapes the specified value, if necessary according to
+     * <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>.
+     *
+     * @param value The value which will be escaped according to
+     *              <a href="https://tools.ietf.org/html/rfc4180#section-2">RFC-4180</a>
+     * @return {@link CharSequence} the escaped value if nesessary, or the value unchanged
+     */
+    public static CharSequence escapeCsv(CharSequence value) {
+        int length = checkNotNull(value, "value").length();
+        if (length == 0) {
+            return value;
+        }
+        int last = length - 1;
+        boolean quoted = isDoubleQuote(value.charAt(0)) && isDoubleQuote(value.charAt(last)) && length != 1;
+        boolean foundSpecialCharacter = false;
+        boolean foundEscapeCharacter = false;
+        StringBuilder escaped = new StringBuilder(length + DEFAULT_NR_OF_DOUBLE_QUOTES).append(DOUBLE_QUOTE);
+        for (int i = 0; i < length; i++) {
+            char current = value.charAt(i);
+            switch (current) {
+                case DOUBLE_QUOTE:
+                    if (i == 0 || i == last) {
+                        if (!quoted) {
+                            escaped.append(DOUBLE_QUOTE);
+                        } else {
+                            continue;
+                        }
+                    } else {
+                        boolean isNextCharDoubleQuote = isDoubleQuote(value.charAt(i + 1));
+                        if (!isDoubleQuote(value.charAt(i - 1)) &&
+                                (!isNextCharDoubleQuote || isNextCharDoubleQuote && i + 1 == last)) {
+                            escaped.append(DOUBLE_QUOTE);
+                        }
+                    }
+                    foundEscapeCharacter = true;
+                case LINE_FEED:
+                case CARRIAGE_RETURN:
+                case COMMA:
+                    foundSpecialCharacter = true;
+            }
+            escaped.append(current);
+        }
+        return foundEscapeCharacter || (foundSpecialCharacter && !quoted) ?
```

Comment:
> this use case is covered by escapeCsvAlreadyQuoted

I was thinking if the input is a properly escaped string with a quote character internally (not first or last character).  For example:

input: `"foo""goo"`
output: `"foo""goo"`

Perhaps I'm missing something but seems like just because you find a `"` doesn't mean that you have to add quotes around the value.